OOP-War/Models/Knight.cs

```diff
@@ -0,0 +1,11 @@
+namespace OOP_War.Models;
+
+public class Knight : Warrior
+{
+    public Knight()
+    {
+        CurrentHealth = 50;
```

Comment:
You override the Health value in derived class, but it is the same as in Warrior class. I think that you need to reuse that value